Update gemrb to 0.9.3

Peppersawce opened the pull request:
Working recipe for building gemrb 0.9.3 on Beta 5 + files needed for it. Build process works but could be better, there's an issue with cmake finding extend2da.py at buildtime that's being fixed in an unorthodox way. Improvements are welcome.

Discussion here: https://discuss.haiku-os.org/t/gemrb-0-9-3-port-questions-about-haikuports-recipes/15535

Begasus commented on games-engines/gemrb/gemrb-0.9.3.recipe:
```diff
@@ -0,0 +1,114 @@
+	"
+
+ARCHITECTURES="all !x86_gcc2"
+SECONDARY_ARCHITECTURES="!x86"
```
Why mark as broken, 32bit hasn't been tested yet :)

Begasus commented on games-engines/gemrb/gemrb-0.9.3.recipe:
```diff
@@ -0,0 +1,114 @@
+{
+	sed -i 's,\/usr/bin/python,\/usr/bin/python3,g' admin/extend2da.py
+}
+BUILD()
```
Empty line between PATCH() and BUILD()

jmairboeck replied:
Actually, `/usr/bin` doesn't make sense on Haiku. Is this for the shebang line? In that case, the path could be removed completely, so just `python3` should work. Otherwise, `/bin/python3` is more correct.

Peppersawce replied:
Yeah, it's for the shebang line

Begasus commented on games-engines/gemrb/gemrb-0.9.3.recipe:
```diff
@@ -0,0 +1,114 @@
+}
+BUILD()
+{
+	sdlheaderdir=$(finddir B_SYSTEM_HEADERS_DIRECTORY)/SDL2
```
for 32bit this probably needs secondaryArchSubDir

Peppersawce replied:
Sounds like a good idea but I have no way to test it on a 32bit system atm and I don't feel like making changes I can't test

Begasus commented on games-engines/gemrb/additional-files/cmake_install.cmake:
```diff
@@ -0,0 +1,96 @@
+  if(CMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION)
+    message(FATAL_ERROR "ABSOLUTE path INSTALL DESTINATION forbidden (by caller): ${CMAKE_ABSOLUTE_DESTINATION_FILES}")
+  endif()
+  file(INSTALL DESTINATION "/packages/gemrb-0.9.3-1/.self/bin" TYPE FILE MESSAGE_NEVER PERMISSIONS OWNER_WRITE OWNER_READ OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE FILES "extend2da.py")
```
Would be nice if the DESTINATION paths would use a variable instead of a hardcoded path

Begasus reviewed (changes requested):
Needs some work, thanks so far!

jmairboeck commented on games-engines/gemrb/additional-files/gemrb.rdef:
```diff
@@ -0,0 +1,36 @@
+resource app_version {
+       major  = 0,
+       middle = 9,
+       minor  = 3,
```
Usually, these are set as "variables" which are replaced with sed in the INSTALL function with values derived from `$portVersion`, and therefore this file has a `.rdef.in` file extension. This is so that it doesn't need to be updated manually for every new version.

You'll find many examples in HaikuPorts how to do this.

jmairboeck commented on games-engines/gemrb/gemrb-0.9.3.recipe:
```diff
@@ -0,0 +1,114 @@
+		-DDATA_DIR=$dataDir/gemrb \
+		-DMAN_DIR=$manDir/man6 \
+		-DSYSCONF_DIR=$settingsDir/gemrb \
+		-DDOC_DIR=$documentationDir/gemrb \
```
Why not use `$docDir`? That includes the package name in the path.

Peppersawce commented:
I reworked the recipe with @Begasus 's feedback and managed to find a solution with sed to fix the build issue cmake has (something about an extra "/")
I'll look around poking with @jmairboeck later as well :)

Peppersawce commented on games-engines/gemrb/additional-files/GemRB.cfg:
```diff
@@ -0,0 +1,283 @@
+
+# Haiku-specific note: Entering GemRB's home folder so you can play from data
+# On a read-only fs. Comment this line if you have r/w access to the data
+SavePath=~/gemrb
```
About that, currently  $settingsDir is the one under system for some reason, which is a less-than-ideal solution as well. I'm honestly slightly confused about where things should go, what would be the proper path(s) where I should store the .cfg and saves in userspace?

Peppersawce replied:
Edited path to "finddir B_USER_SETTINGS_DIRECTORY + /gemrb" as suggested, there's an issue with this path however, the program will not save games if the ~/config/settings/gemrb folder doesn't already exist

Peppersawce commented:
Removed the pre-made cfg file and added sed commands to edit the source's one instead. I've tried editing the config to use better paths but the executable only seems to be able to write in "Home" and I have no clue why tbh

Peppersawce commented on games-engines/gemrb/additional-files/GemRB.cfg:
```diff
@@ -0,0 +1,283 @@
+# This is the path where GemRB will store cached files, enter the full path.
+
+# Haiku-specific note: standard value "./Cache2/" caused all sorts of issues
+CachePath=~/gemrb/cache
```
I agree, but at the moment it seems like the program only has write access to Home, any way to fix that?

Begasus commented:
Thanks so far, I'll let the others comment if anything else should need some ironing.

jmairboeck commented:
> Mmm...
> 
> In `/system/data/cmake/Modules/FindPython/Support.cmake` I see way too many lines that look lie this:
> 
> ```cmake
> list (TRANSFORM suffixes PREPEND "lib/python${version}/config-${version}")
> ```
> 
> And none with "lib/x86/python${version}/config-${version}", so that might explain why it fails on 32 bits.
> 
> So... our cmake is pretty broken right now (in regard to finding Python, at least :-D).
> 
> Do we have someone that understand cmake well enough to fix that?

There should be a CMake variable `${CMAKE_HAIKU_SECONDARY_ARCH_SUBDIR}` (see [here](https://github.com/Kitware/CMake/blob/ac8738c0770667086d43aba024dc749602252011/Modules/Platform/Haiku.cmake#L43)) which could be used here (appended after `lib`).

Begasus commented on games-engines/gemrb/patches/gemrb-0.9.3.patchset:
```diff
@@ -0,0 +1,22 @@
+From d6914f994aab67cd52707da8fda273ba3fe5e6ba Mon Sep 17 00:00:00 2001
+From: Yourself <user@shredder.fritz.box>
```
I think this is done by "hp -e" extracting the patchset, this is set in haikuports.conf file in the line PACKAGER=?

jmairboeck replied:
No, haikuporter uses whatever user information the commits in the work repo have. These aren't created by haikuporter but with regular `git commit`. The PACKAGER in haikuports.conf is just for the metadata of the created packages.

Peppersawce replied:
If it's fine with you all and there's no guideline or anything, I'd rather keep it this way, I get a chuckle out of that kind of headers ngl

Peppersawce commented on games-engines/gemrb/gemrb-0.9.3.recipe:
```diff
@@ -0,0 +1,119 @@
+	rm $settingsDir/gemrb/GemRB.cfg.noinstall.sample
+	rm $settingsDir/gemrb/GemRB.cfg.sample
+
+	# Edit Cache path
+	# Issue: cannot write cache files in most paths, falling back to ~/gemrb/cache
+	#cachedir=$(finddir B_USER_CACHE_DIRECTORY)/gemrb
+	#sed -i "s,.\/Cache2/,${cachedir},g" $settingsDir/gemrb/GemRB.cfg
+	sed -i "s,.\/Cache2/,~\/gemrb/cache,g" $settingsDir/gemrb/GemRB.cfg
```
If that code just makes the directories it's not enough for the cache issue, there's something more to it unfortunately.

It would solve the saves issue though so I will add that to the recipe :)

Peppersawce replied:
Nah I get it, having the cache in home is not ideal and I'd rather have that sorted as well

Peppersawce commented on games-engines/gemrb/gemrb-0.9.3.recipe:
```diff
@@ -0,0 +1,128 @@
+
+PROVIDES="
+	gemrb$secondaryArchSuffix = $portVersion
+	cmd:extend2da.py$secondaryArchSuffix = $portVersion
+	cmd:gemrb$secondaryArchSuffix = $portVersion
```
Let me get this straight: if the recipe doesn't build on gcc2 then $secondaryArchSuffix only goes on the "provides" with the package name and not on cmd: or app: ?
